src/components/TabSelector/TabSelectorBase.tsx

```diff
@@ -0,0 +1,193 @@
+    const [affectedAnimatedTabs, setAffectedAnimatedTabs] = useState(defaultAffectedAnimatedTabs);
+    const viewRef = useRef<View>(null);
+    const [selectorWidth, setSelectorWidth] = useState(0);
+    const [selectorX, setSelectorX] = useState(0);
+
+    const activeIndex = useMemo(() => tabs.findIndex((tab) => tab.key === activeTabKey), [tabs, activeTabKey]);
+
+    // After a tab change, reset affectedAnimatedTabs once the transition is done so
+    // tabs settle back into the default animated state.
+    useEffect(() => {
+        const timerID = setTimeout(() => {
+            setAffectedAnimatedTabs(defaultAffectedAnimatedTabs);
+        }, CONST.ANIMATED_TRANSITION);
+
+        return () => clearTimeout(timerID);
+    }, [defaultAffectedAnimatedTabs, activeIndex]);
+
+    const measure = useCallback(() => {
+        viewRef.current?.measureInWindow((x, _y, width) => {
+            setSelectorX(x);
+            setSelectorWidth(width);
+        });
+    }, []);
+
+    // Measure location/width after initial mount and when layout animations settle.
+    useLayoutEffect(() => {
+        const timerID = setTimeout(() => {
+            measure();
+        }, CONST.TOOLTIP_ANIMATION_DURATION);
+
+        return () => clearTimeout(timerID);
+    }, [measure]);
+
+    // Re-measure when resizing ends so tooltips and equal-width layouts stay aligned.
+    useEffect(() => {
+        if (isResizing) {
+            return;
+        }
+        measure();
+    }, [measure, isResizing]);
+
+    return (
+        <View
+            style={styles.tabSelector}
+            ref={viewRef}
+        >
+            {tabs.map((tab, index) => {
+                const isActive = index === activeIndex;
+                const activeOpacity = getOpacity({
+                    routesLength,
+                    tabIndex: index,
+                    active: true,
+                    affectedTabs: affectedAnimatedTabs,
+                    position,
+                    isActive,
+                });
+                const inactiveOpacity = getOpacity({
+                    routesLength,
+                    tabIndex: index,
+                    active: false,
+                    affectedTabs: affectedAnimatedTabs,
+                    position,
+                    isActive,
+                });
+                const backgroundColor = getBackgroundColor({
+                    routesLength,
+                    tabIndex: index,
+                    affectedTabs: affectedAnimatedTabs,
+                    theme,
+                    position,
+                    isActive,
+                });
+
+                const handlePress = () => {
+                    if (isActive) {
+                        return;
+                    }
+
+                    if (activeIndex >= 0) {
+                        setAffectedAnimatedTabs([activeIndex, index]);
+                    } else {
+                        setAffectedAnimatedTabs([index]);
+                    }
```

Comment:
Earlier it was like this       `setAffectedAnimatedTabs([state.index, index]);`> 

Is this correct Change?